### PR TITLE
Bug 2336 - Add deployment documentation

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,25 +8,18 @@ Supported Platforms
...
+-------------
 
 * node v0.8 or higher
 * npm (comes with node v0.8 installer)
-
-Using Butter with Cornfield
----------------------------
-
-In order for Cornfield to operate correctly ( being able to save, load, and publish ) you will need to install mongodb. You can find a good guide to do so [here](http://www.mongodb.org/display/DOCS/Quickstart). For Cornfield to work correctly, ensure [v2.0.8 or greater](http://www.mongodb.org/downloads) is installed. Before starting up you Cornfield server be sure that you have MongoDB running or else it will fail to start.
+* mongodb v2.0.8 or higher
```
Comment (humphd) [inline README.md:22]: This will need to get updated based on Matt's work, maybe we should just fix that now knowing it's coming?

File: README.md [hunk 2]
```diff
@@ -34,31 +27,58 @@ Environment Setup
 1. `git clone --recursive https://github.com/mozilla/butter.git`
 2. `cd butter`
 3. `npm install`
-4. `node make server`
 
-When updating your repo, make sure to run `node make submodules` to update Butter's submodules.
-
-Running Butter (with cornfield)
+Running Butter
```
Comment (humphd) [inline README.md]: "...in development mode"

File: README.md [hunk 3]
```diff
@@ -34,31 +27,58 @@ Environment Setup
...
-
-Running Butter (without cornfield)
-----------------------------------
-
-* Be sure to have a webserver of some sort running and navigate to butter/templates/basic/ ( keep in mind you will not be able to save and load projects )
-* Run `node make` to ensure all of the CSS has been compiled, since Stylus files are used in place of plain CSS
-* If you do not have a webserver setup you can create a temporary one by running `python -m SimpleHTTPServer 8888` inside the root of your Butter directory
-* Navigate to localhost:8888 to use butter
+1. Run `node make server`.
+2. Navigate to http://localhost:8888/
```
Comment (humphd) [inline README.md]: Add a note about how we bind IP/port in the cornfield config files, people always ask.

File: README.md [hunk 4]
```diff
@@ -34,31 +27,58 @@ Environment Setup
...
 
 Packaging and Distributing Butter
 --------------------------------
 
-Running `node make deploy` will compile all the necessary files into the `dist/` folder, including resources like editors and dialogs which comprise a complete running environment for Butter.
-
-Within `dist/` you will find `src/butter.js` and `external/popcorn-js/popcorn.js`, which are single-file versions of the Butter and Popcorn.js sources, customized to the basic template.  You can find these templates in `dist/templates/basics`.
+Running `node make deploy` will compile all the necessary files into the `dist/` folder.
+Run `NODE_ENV=production node app.js` in the cornfield folder to start the server.
```
Comment (humphd) [inline README.md]: "in the dist/cornfield directory in order to run the server in production mode"

File: README.md [hunk 5]
```diff
@@ -34,31 +27,58 @@ Environment Setup
...
+    "format" : "default"
+  },
+  "session" : {
+    "secret": "1721f7a15316469fa4a9-5117d0d20e9f"
+  },
+  "staticMiddleware": {
+    "maxAge": "3600000"
+  },
+  "dirs": {
+    "hostname": "http://example.org"
```
Comment (humphd) [inline README.md:85]: remind me when you review the AWS S3 stuff that we have to add more info here...

File: README.md [hunk 6]
```diff
@@ -34,31 +27,58 @@ Environment Setup
...
+
+There are several configuration files in cornfield/config/ that control how cornfield works.
+They are applied in order from most general to most specific to present one configuration
+to the server:
+
+1. default.json
+2. _hostname_.json
+3. _environment_.json
+4. _hostname_-_environment_.json
+5. runtime.json
```
Comment (humphd) [inline README.md:55]: You should mention what each of these are, you only really deal with one of them.

File: README.md [hunk 7]
```diff
@@ -66,12 +86,7 @@ Testing
 Before contributing a new patch be sure to run the following:
 
 * `node make check` to lint butter
-* run the qUnit tests in the test/ directory to be sure you broke nothing
-
-Documentation
--------------
-
-Documentation can be created from the source files by running `node make docs`. Running this command will create `/doc` and fill it with markdown files corresponding to documentation found within the source.
+* run the QUnit tests in the test/ directory to be sure you broke nothing
```
Comment (humphd) [inline README.md]: Should mention that these require you to run 'node make server' first, and then navigate to localhost:8888/test

File: README.md (hunk 1 above)
Comment (jbuck) [inline README.md:22]: I figure we'll just change this requirement (mysql or sqlite3) in Matt's patch when that gets closer to landing

File: README.md (hunk 3 above)
Comment (jbuck) [inline README.md]: Done below now

File: README.md (hunk 5 above)
Comment (jbuck) [inline README.md:85]: Will do

File: README.md (hunk 6 above)
Comment (jbuck) [inline README.md:55]: Added below

File: README.md [hunk 8]
```diff
@@ -8,25 +8,18 @@ Supported Platforms
 
 We're writing Butter so that it runs in modern, HTML5 compatible browsers.  For version 1.0, we're targeting modern HTML5 desktop browsers.  In the current version we are supporting:
 
-#### Desktop (Windows, OS X, Linux):
-* Firefox stable
+### Desktop:
 * Chrome stable
+* Firefox stable
+* Internet Explorer 9
```
Comment (secretrobotron) [inline README.md]: Not more recent versions of IE?

File: README.md [hunk 9]
```diff
@@ -89,7 +109,7 @@ Contributing
 # Get started working on Butter
 
 1. Join the #popcorn IRC channel
-A lot of the people working on Butter are in the #popcorn channel on irc.mozilla.org during business hours M to F (most of us are in Toronto, so Eastern Time). 
+A lot of the people working on Butter are in the #popcorn channel on irc.mozilla.org during business hours M to F (most of us are in Toronto, so Eastern Time).
```
Comment (secretrobotron) [inline README.md]: This doesn't format properly: https://github.com/jbuck/butter/tree/t2336

Comes out as "Join the #popcorn IRC channel A lot of people..."

**Pull request merged**